### PR TITLE
Introduce a short-hand notation for strict versions

### PR DESCRIPTION
**Pull request by melix**
### Context

Polishing of 6.0 features. Also includes improved docs for declaration of versions.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* A Maven-style version range: e.g. `[1.0,)`, `[1.1, 2.0)`, `(1.2, 1.5]`
+** The `[` and `]` symbols indicate an inclusive bound; `(` and `)` indicate an exclusive bound.
+** When the upper or lower bound is missing, the range has no upper or lower bound.
+** The symbol `]` can be used instead of `(` for an exclusive lower bound, and `[` instead of `)` for exclusive upper bound. e.g `]1.0, 2.0[`
+* A _prefix_ version range: e.g. `1.+`, `1.3.+`
+** Only versions exactly matching the portion before the `+` are included.
+** The range `+` on it's own will include any version.
+* A `latest-status` version: e.g. `latest.integration`, `latest.release`
+** Will match the highest versioned module with the specified status. See link:{javadocPath}/org/gradle/api/artifacts/ComponentMetadata.html#getStatus--[ComponentMetadata.getStatus()].
+* A Maven `SNAPSHOT` version identifier: e.g. `1.0-SNAPSHOT`, `1.4.9-beta1-SNAPSHOT`
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc]: I think the new strict version notation "!!" should also be in this list, wdyt?

Comment (melix) [reply subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc]: mmm, I'm not sure. This notation is described below. The idea was to describe:

- the simplest declaration
- then semantics of ordering
- then semantics of strict (and the short-hand notation)

File: subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Versions are ordered based on the following rules:
+
+* Each version is split into it's constituent "parts":
+** The characters `[. - _ +]` are used to separate the different "parts" of a version.
+** Any part that contains both digits and letters is split into separate parts for each: `1a1 == 1.a.1`
+** Only the parts of a version are compared. The actual separator characters are not significant: `1.a.1 == 1-a+1 == 1.a-1 == 1a1`
+* The equivalent parts of 2 versions are compared using the following rules:
+** If both parts are numeric, the highest numeric value is **higher**: `1.1` < `1.2`
+** If one part is numeric, it is considered **higher** than the non-numeric part: `1.a` < `1.1`
+** If both are not numeric, the parts are compared **alphabetically, case-sensitive**: `1.a` < `1.A` < `1.b` < `1.B`
```
[the remark this answers is not in the document]
Comment (melix) [reply subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc]: No, this is compared alphabetically _then_ by case sensitivity.

[the remark this answers is not in the document]
Comment (melix) [reply subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc]: mmm, actually you're right, the javadocs were wrong...

File: subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+** A version with an extra numeric part is considered **higher** than a version without: `1.1` < `1.1.0`
+** A version with an extra non-numeric part is considered **lower** than a version without: `1.1` < `1.1.a`
```
[the remark this answers is not in the document]
Comment (melix) [reply subprojects/docs/src/docs/userguide/dep-man/02-declaring-dependency-versions/single_versions.adoc]: good catch! Also fixing the javadocs...

**Pull request merged**